### PR TITLE
[Task 1470] Part 1: Add Basic Rendering Test + Mocks

### PR DESCRIPTION
**Pull request by msa2984**
**Work Item**:
https://dev.azure.com/csebostoncrew/Retrospectives-AzDO-Extension/_boards/board/t/Retrospectives-AzDO-Extension%20Team/Stories/?workitem=1470

**Notes**:
This PR only includes a simple test for verifying the mocked FeedbackItem props are used to set values in the rendered FeedbackItem. I will follow up with a more in-depth PR which includes more specific tests, however I wanted to make sure that the mocks that Phong figured out for `azure-devops-extension-api` and `azure-devops-extension-sdk` were extracted out in files which can be used in multiple tests.

### CODE REVIEW

File: RetrospectiveExtension.Frontend/components/feedbackItem.tsx
Comment (msa2984) [inline RetrospectiveExtension.Frontend/components/feedbackItem.tsx:834]: it looks like a lot of this file was changed - however all I did was move the ...Hidden states into the `<Dialog>` to set the hidden value. 

For example:

```
 {!this.state.isDeleteItemConfirmationDialogHidden &&
  <Dialog
     hidden={false}>...
  </Dialog>
}
```
became: 
```
<Dialog
   hidden={this.state.isDeleteItemConfirmationDialogHidden }>...
</Dialog>
```

File: RetrospectiveExtension.Frontend/components/__tests__/feedbackItem.test.tsx
Comment (msa2984) [inline RetrospectiveExtension.Frontend/components/__tests__/feedbackItem.test.tsx:175]: so after I merged in the application insights changes from dev, all of my expects started failing. adding this .dive() call resolved the issue - the `withAITracking` wrapper added to the components makes enzyme think the wrapper is the top level component, so the shallow render believes the first level of children is the `FeedbackItem`.

**Review by polatengin — APPROVED** [review]
LGTM! thanks 🚀💯

File: CONTRIBUTING.md
Comment (NarmathaBala) [inline CONTRIBUTING.md]:
```suggestion
Components functionality is tested using the [Jest Testing Framework](https://jestjs.io/), and the
```

File: CONTRIBUTING.md
Comment (NarmathaBala) [inline CONTRIBUTING.md]:
```suggestion
[Enzyme Testing utility](https://enzymejs.github.io/enzyme/) and
```

**Review by NarmathaBala — APPROVED** [review]
Couple of text suggestions! Otherwise looks good. Thanks @msa2984

File: RetrospectiveExtension.Frontend/components/feedbackItem.tsx
Comment (phongcao) [reply RetrospectiveExtension.Frontend/components/feedbackItem.tsx:834]: Hey, I really appreciate these kinds of comment. Really help reviewers to check the code faster.

**Comment by phongcao** [comment]
@msa2984 @focusthenflex Such a great PR, awesome work! I'm late to this one but there is a small suggestion that we could do for future updates. All the files under mock folder should use `2-space` indentation to be consistent with the rest of files.

File: RetrospectiveExtension.Frontend/tsconfig.json
Comment (phongcao) [inline RetrospectiveExtension.Frontend/tsconfig.json:3]: @msa2984 @focusthenflex Why did you add this config? Is it to fix the error with rendering feedback item?

**Comment by msa2984** [comment]
> @msa2984 @focusthenflex Such a great PR, awesome work! I'm late to this one but there is a small suggestion that we could do for future updates. All the files under mock folder should use `2-space` indentation to be consistent with the rest of files.

ah, my bad Phong - I have another review I'm going to be starting in the next day or so to add more tests for the feedbackItem, so I'll address formatting there

File: RetrospectiveExtension.Frontend/tsconfig.json
Comment (msa2984) [reply RetrospectiveExtension.Frontend/tsconfig.json:3]: hmm... so from my notes, I see that the rendering issues were fixed when adding the "strictNullChecks" option -also, I think this shouldn't matter anyways since we're using a separate tsconfig for testing anyways... so this change should be reverted

Comment (phongcao) [reply RetrospectiveExtension.Frontend/tsconfig.json:3]: We actually use it for creating the bundle file in production as well. If you try removing this flag on our  current`dev` branch and test the production bundle (`npm run build:p`), there will be an issue when creating feedback item. We can talk more about it in our stand up but it's good for now.

**Pull request merged**
